community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/internal/BatchInserterImpl.java

```diff
@@ -443,47 +443,48 @@ private void repopulateAllIndexes() throws IOException, IndexEntryConflictExcept
         }
 
         final IndexRule[] rules = getIndexesNeedingPopulation();
-        final IndexPopulator[] populators = new IndexPopulator[rules.length];
+        final Map<LabelSchemaDescriptor,IndexPopulator> populators = new HashMap<>();
+        final Map<LabelSchemaDescriptor,NewIndexDescriptor> indexes = new HashMap<>();
         // the store is uncontended at this point, so creating a local LockService is safe.
 
-        final NewIndexDescriptor[] descriptors = new NewIndexDescriptor[rules.length];
+        final LabelSchemaDescriptor[] descriptors = new LabelSchemaDescriptor[rules.length];
 
         for ( int i = 0; i < rules.length; i++ )
         {
             IndexRule rule = rules[i];
-            descriptors[i] = rule.getIndexDescriptor();
-            populators[i] = schemaIndexProviders.apply( rule.getProviderDescriptor() )
-                                                .getPopulator( rule.getId(), descriptors[i], new IndexSamplingConfig( config ) );
-            populators[i].create();
+            NewIndexDescriptor index = rule.getIndexDescriptor();
+            descriptors[i] = index.schema();
+            IndexPopulator populator = schemaIndexProviders.apply( rule.getProviderDescriptor() )
+                                                .getPopulator( rule.getId(), index, new IndexSamplingConfig( config ) );
+            populator.create();
+            populators.put( index.schema(), populator );
+            indexes.put( index.schema(), index );
         }
 
         Visitor<NodeUpdates, IOException> propertyUpdateVisitor = updates -> {
             // Do a lookup from which property has changed to a list of indexes worried about that property.
-            for ( int i = 0; i < descriptors.length; i++ )
+            for ( IndexEntryUpdate indexUpdate :
+                    updates.forIndexes( Iterables.asIterable( descriptors ), NO_UNCHANGED_PROPERTIES ) )
```

Comment:
Again, it would be nice to note _why_ `NO_UNCHANGED_PROPERTIES` is appropriate here.